src/query/functions/aggregation/take.go

```diff
@@ -198,3 +274,28 @@ func takeFn(heap utils.FloatHeap, values []float64, buckets [][]int) []float64 {
 
 	return values
 }
+
+func takeInstantFn(heap utils.FloatHeap, values []float64, buckets [][]int, metas []block.SeriesMeta) []valueAndMeta {
+	var result []valueAndMeta
+	if heap.Cap() < 1 {
+		return result
```

Comment:
This may need to nil out values; `util.Memset(values, math.NaN())`

Reply:
There is no need for this because we are not mutating `values` in this method or using them afterwards.